### DISCUSSION TITLE
Multiple installations - package already installed

### DISCUSSION THREAD

**Discussion by anentropic** [Homebrew/discussions#300] [Everyday usage]
I am still feeling my way how best to use Homebrew on a new M1 macbook...

The docs here suggest you can have multiple homebrew installations:
https://docs.brew.sh/Installation#multiple-installations
(as linked from https://github.com/mikelxc/Workarounds-for-ARM-mac)

> do yourself a favour and install to `/usr/local` on macOS Intel, `/opt/homebrew` on macOS ARM

and

> Whichever brew command is called is where the packages will be installed.

And a few sources suggest creating a duplicate of Terminal.app and setting one of them to run under Rosetta emulation
https://www.notion.so/Run-x86-Apps-including-homebrew-in-the-Terminal-on-Apple-Silicon-8350b43d97de4ce690f283277e958602
https://stackoverflow.com/a/64885034/202168

So this is what I've done... I used the Rosetta-fied terminal to install Homebrew as if on an Intel mac, and I used a regular (arm) Terminal to install a second Homebrew to `/opt/homebrew` and then symlinked `/opt/homebrew/bin/brew -> /usr/local/bin/armbrew`

So far so good, seemingly.

But if I try to install packages under `armbrew` that are already installed in regular `brew` it just says they're already installed.

Is there some way to be able to have separate brew installs of the same package, but built for different archs?

**Comment by carlocab** [comment]
There is, but not when the ARM `brew` is symlinked to `/usr/local`. Any `brew` run from there expects to be on Intel/Rosetta.

**Reply by anentropic** [reply]
so instead of a symlink... if I create an alias instead, would that work?

**Reply by anentropic** [reply]
(I'm trying to avoid having to `cd /opt/homebrew/bin` first every time)

**Reply by carlocab** [reply]
As in `alias armbrew='/opt/homebrew/bin/brew'`? Yes, I believe that should work. `armbrew doctor` may complain about it not being in your `PATH`.

**Comment by gromgit** [comment]
I don't have an M1 to test it on, but it strikes me that this is something `brew shellenv` should (eventually) be dealing with. During this transition period, you can probably work around this with the following in your `.bashrc`:
```
if rosetta_is_active; then
  eval "$(/usr/local/bin/brew shellenv)"
else
  eval "$(/opt/homebrew/bin/brew shellenv)"
fi
```
Someone else would have to advise on how to implement `rosetta_is_active`, i.e. how to test if this is running in a Rosetta-fied bash.
